### DISCUSSION TITLE
Favorite dashboards access

### DISCUSSION THREAD

**Discussion by amitmiran137** [apache/superset#12839] [Ideas]
this topic was brought up in 
https://github.com/apache/superset/pull/12680/files/2c2ade827f8f9bfb089efab1c7e3a143fa691fe5#r566474303

we need to re-evaluate the logic behind allowing any favorite dashboard to be accessed by the user without any basic access control like the rbac and the data level 

https://github.com/apache/superset/blob/master/superset/dashboards/filters.py#L47

@villebro @dpgaspar @betodealmeida @ktmud

**Comment by ktmud** [comment]
Another direct link for reference: https://github.com/apache/superset/pull/12680#discussion_r566782036

I think if users can favorite the dashboard, it means they have at least had access to it at some point before. It would be nice if we still display the dashboard in the list, but tell them somewhere they have lost the access (either in the list or when opening the dashboard).

cc @junlincc @mihir174
